src/main.js

```diff
@@ -0,0 +1,171 @@
+    NEXT_LEVEL_BTN: 'next-level-button',
+    NEW_GAME_BTN: 'new-game-button',
+    NEXT_LEVEL: 'next-level',
+};
+
+const holes = document.querySelectorAll(selectors.HOLE);
+const scoreBoardElement = document.querySelector(selectors.SCORE);
+const modalWindowElement = document.querySelector(selectors.MODAL);
+const hole7Element = document.querySelector(selectors.HOLE_7);
+const hole8Element = document.querySelector(selectors.HOLE_8);
+let score = 0;
+let minMoleDisappearanceRate;
+let maxMoleDisappearanceRate;
+let timerId;
+let lastHole;
+let isGameFinished = false;
+
+function getRandomTime(min, max) {
+    return Math.round(Math.random() * (max - min) + min);
+}
+
+function getRandomHole(holes) {    
+    const holesRandomElementIndex = Math.floor(Math.random() * holes.length);
+    const newHole = holes[holesRandomElementIndex];    
+
+    if (newHole === lastHole) {
+        return getRandomHole(holes);
+    }
+    lastHole = newHole;
+    return newHole;  
+}
+
+function peepMole() {
+    const time = getRandomTime(minMoleDisappearanceRate, maxMoleDisappearanceRate);
+    const hole = getRandomHole(holes);
+
+    if(!isGameFinished) {
+        hole.classList.add(classes.UP);
+        timerId = setTimeout(() => {
+            hole.classList.remove(classes.UP);
+            peepMole();
+        }, time);         
+    }        
+}
+
+function finishGame() {
+    isGameFinished = true;
+    document.querySelectorAll(selectors.UP).forEach((el) => el.classList.remove(classes.UP));
+    localStorage.setItem('score', score);
+    localStorage.setItem('min', minMoleDisappearanceRate);
+    localStorage.setItem('max', maxMoleDisappearanceRate);
+    clearTimeout(timerId);
+    modalWindowElement.classList.add(classes.ACTIVE);
+}
+
+function startLevel() {
+    isGameFinished = false;
+    peepMole();
+    setTimeout(finishGame, 10000);
+}
+
+function startGame() {    
+    score = localStorage.getItem('score') || 0;
+    minMoleDisappearanceRate = localStorage.getItem('min') || 200;
+    maxMoleDisappearanceRate = localStorage.getItem('max') || 3000;
+    startLevel();     
+}
+
+function whackMole(e) {
+    if(!e.isTrusted) return;
+    if (e.target.className.includes(classes.MOLE)) {
+        score++; 
+        e.target.parentNode.classList.remove(classes.UP);
+        scoreBoardElement.textContent = score;
+    }    
+}
+
+function closeModalWindow() {
+    modalWindowElement.classList.remove(classes.ACTIVE);
+}
+
+function repeatLevel() {
+    closeModalWindow(); 
+    startLevel();
+}
+
+function goToNextLevel() {
+    minMoleDisappearanceRate *= 0.8;
+    maxMoleDisappearanceRate *= 0.7;
+    closeModalWindow();
+    holes.forEach((el) => el.classList.add(classes.NEXT_LEVEL));
+    hole7Element.classList.add(classes.ACTIVE);    
+    hole8Element.classList.add(classes.ACTIVE);
+    startLevel();    
+}
+
+function startNewGame() {
+    localStorage.removeItem('score');
+    localStorage.removeItem('min');
+    localStorage.removeItem('max');
```

Comment:
let's move to a separate function - cleanUpLevelDataFromLocalStorage or smth like this